helm-app-operator/pkg/helm/helm.go

```diff
@@ -44,6 +60,23 @@ func NewInstaller(storageBackend *storage.Storage, tillerKubeClient *kube.Client
 	return installer{storageBackend, tillerKubeClient, chartDir}
 }
 
+// NewInstallerFromEnv returns a GVK and installer based on configuration provided
+// in the environment.
+func NewInstallerFromEnv(storageBackend *storage.Storage, tillerKubeClient *kube.Client) (schema.GroupVersionKind, Installer, error) {
+	apiVersion := os.Getenv(APIVersionEnvVar)
+	kind := os.Getenv(KindEnvVar)
+	chartDir := os.Getenv(HelmChartEnvVar)
```

Comment:
I know the changes form this PR are temporary, but these envs could be unset.
Not sure what happens in that case with an nil valued gvk.

Reply:
Good catch. The controller eventually panicked trying to add the watches, which I think was the behavior before as well. Updated with some validation.